lib/puppet/provider/package/yum.rb

```diff
@@ -159,6 +172,11 @@ def install
         self.debug "Downgrading package #{@resource[:name]} from version #{is[:ensure]} to #{should}"
         operation = :downgrade
       end
+
+      if is && rpm_compareEVR(rpm_parse_evr(should), rpm_parse_evr(is[:ensure])) > 0
+        self.debug "Updrading package #{@resource[:name]} from version #{is[:ensure]} to #{should}"
```

Comment:
Not sure, I just followed the pattern from the downgrade block, but that's fairly old (9ccd29f3aacdab03f2ea9a693b5bca928439683b, 2010 commit!) so can remove if needed?